controllers/clusterresources/awsvpcpeering_controller.go

```diff
@@ -338,30 +403,75 @@ func (r *AWSVPCPeeringReconciler) startAWSVPCPeeringStatusJob(awsPeering *v1beta
 func (r *AWSVPCPeeringReconciler) newWatchStatusJob(awsPeering *v1beta1.AWSVPCPeering) scheduler.Job {
 	l := log.Log.WithValues("component", "AWSVPCPeeringStatusJob")
 	return func() error {
-		instaPeeringStatus, err := r.API.GetPeeringStatus(awsPeering.Status.ID, instaclustr.AWSPeeringEndpoint)
+		ctx := context.Background()
+
+		namespacedName := client.ObjectKeyFromObject(awsPeering)
+		err := r.Get(ctx, namespacedName, awsPeering)
+		if k8serrors.IsNotFound(err) {
+			l.Info("Resource is not found in the k8s cluster. Closing Instaclustr status sync.",
+				"namespaced name", namespacedName)
+
+			go r.Scheduler.RemoveJob(awsPeering.GetJobID(scheduler.StatusChecker))
+
+			return nil
+		}
+
+		instaAWSPeering, err := r.API.GetAWSVPCPeering(awsPeering.Status.ID)
 		if err != nil {
-			l.Error(err, "cannot get AWS VPC Peering Status from Inst API", "AWS VPC Peering ID", awsPeering.Status.ID)
+			if errors.Is(err, instaclustr.NotFound) {
+				l.Info("The resource has been deleted on Instaclustr, deleting resource in k8s...")
+				return r.Delete(ctx, awsPeering)
+			}
+
+			l.Error(err, "cannot get AWS VPC Peering StatusCode from Inst API", "AWS VPC Peering ID", awsPeering.Status.ID)
 			return err
 		}
 
-		if !arePeeringStatusesEqual(instaPeeringStatus, &awsPeering.Status.PeeringStatus) {
+		instaPeeringStatus := v1beta1.PeeringStatus{
+			ID:         instaAWSPeering.ID,
+			StatusCode: instaAWSPeering.StatusCode,
+		}
+
+		if !arePeeringStatusesEqual(&instaPeeringStatus, &awsPeering.Status.PeeringStatus) {
 			l.Info("AWS VPC Peering status of k8s is different from Instaclustr. Reconcile statuses..",
 				"AWS VPC Peering Status from Inst API", instaPeeringStatus,
 				"AWS VPC Peering Status", awsPeering.Status)
 
 			patch := awsPeering.NewPatch()
-			awsPeering.Status.PeeringStatus = *instaPeeringStatus
-			err := r.Status().Patch(context.Background(), awsPeering, patch)
+			awsPeering.Status.PeeringStatus = instaPeeringStatus
+			err := r.Status().Patch(ctx, awsPeering, patch)
 			if err != nil {
 				return err
 			}
 		}
 
+		if awsPeering.Annotations[models.UpdateQueuedAnnotation] == "" &&
```

Comment:
how about using the `UpdatingEvent` annotation?

Reply:
It makes sense. Good catch, thanks. Fixed